features/step_definitions/book_index_page_steps.rb

```diff
@@ -7,8 +9,7 @@
 end
 
 Then(/^the books are ordered by published date$/) do
-  expect(Book.first).to eq Book.order(:published_date).first
-  expect(Book.last).to eq Book.order(:published_date).last
+  expect(Book.first).to eq Book.order(published_date: :desc).first
```

Comment:
Here, this is just checking the database. You're not actually checking anything on the page.